test/integration/endpoints_test.go

```diff
@@ -0,0 +1,447 @@
+		assert.Equal(t, float64(2), status["mcp_servers"]) // 2 MCP servers
+		assert.Contains(t, status, "address")
+	})
+
+	t.Run("Internal Config Endpoint", func(t *testing.T) {
+		resp, err := client.Get(baseURL + "/_internal/config")
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+		var config map[string]interface{}
+		err = json.NewDecoder(resp.Body).Decode(&config)
+		require.NoError(t, err)
+
+		assert.Contains(t, config, "server")
+		assert.Contains(t, config, "providers")
+		assert.Contains(t, config, "mcp")
+
+		// Verify API keys are sanitized (not included)
+		providers := config["providers"].([]interface{})
+		for _, provider := range providers {
+			providerMap := provider.(map[string]interface{})
+			assert.NotContains(t, providerMap, "api_key")
+		}
+	})
+
+	t.Run("Internal Metrics Endpoint", func(t *testing.T) {
+		resp, err := client.Get(baseURL + "/_internal/metrics")
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+		var metrics map[string]interface{}
+		err = json.NewDecoder(resp.Body).Decode(&metrics)
+		require.NoError(t, err)
+
+		assert.Contains(t, metrics, "requests_total")
+		assert.Contains(t, metrics, "message")
+	})
+
+	t.Run("Backward Compatibility - Old Models Endpoint", func(t *testing.T) {
+		resp, err := client.Get(baseURL + "/v1/models")
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+		var models map[string]interface{}
+		err = json.NewDecoder(resp.Body).Decode(&models)
+		require.NoError(t, err)
+
+		assert.Equal(t, "list", models["object"])
+		data := models["data"].([]interface{})
+		assert.Len(t, data, 3) // Same models as new endpoint
+	})
+
+	t.Run("Invalid Endpoint Returns 404", func(t *testing.T) {
+		resp, err := client.Get(baseURL + "/invalid/endpoint")
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		assert.Equal(t, http.StatusNotFound, resp.StatusCode)
+	})
+}
+
+// TestIntegration_SocketSecurity tests socket mode security restrictions
+func TestIntegration_SocketSecurity(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	// Create test config
+	cfg := &config.Config{
+		Providers: []config.Provider{
+			{
+				Name:     "test",
+				Type:     "openai",
+				BaseURL:  "http://localhost:8080",
+				APIKey:   "test-key",
+				Models:   []string{"test-model"},
+				Priority: 1,
+			},
+		},
+		Server: config.Server{
+			LogLevel:       "info",
+			MaxRequestSize: 1024,
+		},
+	}
+
+	// Create temporary socket
+	tmpDir := t.TempDir()
+	socketPath := tmpDir + "/security-test.socket"
+
+	// Start socket server
+	srv := server.NewWithSocket(cfg, socketPath)
+
+	go func() {
+		if err := srv.Start(); err != nil && err != http.ErrServerClosed {
+			t.Logf("Server error: %v", err)
+		}
+	}()
+
+	// Give server time to start
+	time.Sleep(200 * time.Millisecond)
```

Comment:
ew, don't sleep

Reply:
+1, let's avoid using sleep in tests whenever possible.